File: queue using stack.py
# A FIFO method that apends 'push' method elements to the first stack 
#while 'pop' and 'peek' methods transfer elements from the first stack to the second stack 
#empty method checks if both stacks are empty, indicating an empty queue.

class MyQueue(object):

    def __init__(self):
        self.in_stack = []
        self.out_stack = []

    def push(self, x):
        self.in_stack.append(x)

    def pop(self):
        self.peek()
        return self.out_stack.pop()

    def peek(self):
        if not self.out_stack:
            while self.in_stack:
                self.out_stack.append(self.in_stack.pop())
        return self.out_stack[-1]

    def empty(self):
        return not self.in_stack and not self.out_stack



queue = MyQueue()

queue.push(1)
queue.push(2)
queue.push(3)

print(queue.pop())  
print(queue.peek())  
print(queue.empty())  

    
# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: Queue_using_Stack.java
import java.util.Stack;

class MyQueue {
    Stack<Integer> in;
    Stack<Integer> out;

    public MyQueue() {
        this.in = new Stack<Integer>();
        this.out = new Stack<Integer>();
    }
    
    public void push(int x) {
        in.push(x);
    }
    
    
    public int pop() {
        peek();
        return out.pop();
    }
    
    public int peek() {
        if(out.isEmpty()){
            while(!in.isEmpty()){
                int popped = in.pop();
                out.push(popped);
            }
        }
        return out.peek();
    }
    
    public boolean empty() {
        if(in.isEmpty() && out.isEmpty()){
            return true;
        }
        return false;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */